### ISSUE TITLE
domain_id failure when authenticating as ldap user

### ISSUE THREAD

**Issue by jedrington** [SUSE-Cloud/keystone-hybrid-backend#17]
Hi there,

I've upgraded to the latest commit of keystone-hybrid-backend in icehouse branch and I'm no longer able to authenticate as an ldap user (sql user still works). The error message in the log file indicates a key error with domain_id [1]. This appears to have been introduced by the fix for issue #12. When I commented out these two lines [2], ldap authentication resumed working.

My ldap users authenticate to the default domain, although I do have a second domain created for heat per the recommendation here: https://wiki.openstack.org/wiki/ReleaseNotes/Icehouse#Stack_domain_users. I'm guessing having multiple domains enabled, while assuming ldap authentication does not require a domain as mentioned in the comments, is what is causing the domain_id error.

Any thoughts?

Thanks,

John

[1]:

2014-09-25 17:44:50.612 30353 DEBUG keystone.common.ldap.core [-] LDAP unbind unbind_s /usr/lib/python2.7/site-packages/keystone/common/ldap/core.py:777
2014-09-25 17:44:50.616 30353 ERROR keystone.common.wsgi [-] 'domain_id'
2014-09-25 17:44:50.616 30353 TRACE keystone.common.wsgi Traceback (most recent call last):
2014-09-25 17:44:50.616 30353 TRACE keystone.common.wsgi   File "/usr/lib/python2.7/site-packages/keystone/common/wsgi.py", line 212, in **call**
2014-09-25 17:44:50.616 30353 TRACE keystone.common.wsgi     result = method(context, **params)
2014-09-25 17:44:50.616 30353 TRACE keystone.common.wsgi   File "/usr/lib/python2.7/site-packages/keystone/token/controllers.py", line 101, in authenticate
2014-09-25 17:44:50.616 30353 TRACE keystone.common.wsgi     core.validate_auth_info(self, user_ref, tenant_ref)
2014-09-25 17:44:50.616 30353 TRACE keystone.common.wsgi   File "/usr/lib/python2.7/site-packages/keystone/token/core.py", line 78, in validate_auth_info
2014-09-25 17:44:50.616 30353 TRACE keystone.common.wsgi     user_ref['domain_id'])
2014-09-25 17:44:50.616 30353 TRACE keystone.common.wsgi KeyError: 'domain_id'
2014-09-25 17:44:50.616 30353 TRACE keystone.common.wsgi

[2]:

``` python
    def is_domain_aware(self):
        # XXX we only need domain_aware to be False when authenticating
        # as an LDAP user; after that, all operations will be done on
        # the SQL database and domain_aware needs to be True. This code
        # makes the assumption that the result of authenticate() should
        # be read as not domain_aware (for LDAP), after which
        # domain_aware should revert to True.
        domain_aware = self.domain_aware
        #if not self.domain_aware:
        #    self.domain_aware = True
        return domain_aware
```

**Comment by symonds** [comment]
I am currently seeing the same issue with the latest version and only a single LDAP domain.

**Comment by sergio97** [comment]
Commenting-out those 2 lines breaks creating a new user, as expected. However, it would fix this and also issue #20 (which, now that I look into it, is a very similar problem to this one).
But I think we could figure out a solution that doesn't break creating a user. I'll keep working on it

**Comment by rhafer** [comment]
This is a duplicate of #20. Closing
